### DISCUSSION TITLE
Accepting code piece by piece

### DISCUSSION THREAD

**Discussion by megasteve19** [community/community#7363] [Copilot Conversations]
Hi! For a day I have been using Copilot with VS Code. I can say It's freaking awesome! But there's something that annoys me; large code chunk generated by pressing tab. Here's the suggestion; We could have a option that allows us to generate code by piece, I press tab Copilot gives me a line, I press tab and again Copilot gives me a line.. And It's goes like that.

**Comment by Urethramancer** [comment]
Many have the opposite problem - sometimes Copilot only suggests one line at a time, when it would be better to just output the whole function. I see that in some boilerplate setup code, or switch statements. So what they need to do is make the behaviour configurable and consistent.

**Comment by uBaby4life** [comment]
If you want something like one liner code then u should be using tabnine and it something like the way you've described. Give you one line of code instead of generating the whole thing.

**Reply by megasteve19** [reply]
No, that's not what I mean. As I said It's awesome, I used tabnine as well. It's not like Copilot. What I mean is; Copliot gives me a suggestion, but some lines of code is no needed.

**Comment by MulleDK19** [comment]
It already does that as long as you're not completing a scope.
If you're inside a function, it'll give you line by line.
